### PR TITLE
feat: implement /feedback command with context generation and URL sharing

### LINKED ISSUE #92: Feedback URL
Allow users to provide feedback via a /feedback command.

Link should point to a Google Form to collect feedback.

### PR DESCRIPTION
**Pull request by btimothy-har**
## Summary

Implements a comprehensive `/feedback` command that allows users to submit feedback about their conversation experience through a Google Forms integration. This addresses the need for users to provide input on bot performance and conversation quality.

**Closes #92**

### Key Features

- **Smart Context Generation**: Uses LLM agent to anonymize and summarize conversation context
- **UUID Tracking**: Each feedback submission gets a unique identifier for tracking
- **Google Forms Integration**: Pre-fills form parameters with context, environment, and version info
- **URL Shortening**: Uses TinyURL to create user-friendly links
- **Session Support**: Works both with and without active chat sessions
- **Caching**: 5-minute TTL cache prevents redundant context generation
- **Background Processing**: Context generation runs asynchronously to avoid blocking user interaction

#### Core Implementation
- **New `/feedback` command handler** with comprehensive error handling and user experience features
- **New LLM agent** (`agent_feedback_context`) for generating anonymized conversation summaries
- **UUID-based tracking** for feedback submissions with Google Forms integration
- **TTL caching system** to optimize context generation performance

#### Data Model Updates  
- **Updated `CommandUsage` model** to support nullable `session_id` for feedback without active sessions
- **Enhanced `ChatEvent` model** with better `__lt__` comparison for sorting events

#### Application Integration
- **Registered feedback command** in bot setup and command constants
- **Background context generation** in message handlers for proactive feedback preparation
- **URL shortening utility** integration for user-friendly feedback links

#### Test Coverage
- **Comprehensive test suite** with 15+ test cases covering all scenarios:
  - Feedback with/without active sessions
  - Context generation, caching, and cache expiration
  - URL formatting and parameter inclusion
  - Agent configuration and dependencies
  - Error handling and edge cases
- **Enhanced test fixtures** for better test data management
- **Full mocking** of external dependencies (database, LLM, URL shortening)

### Technical Details

- **Google Forms URL**: Pre-populated with UUID, session ID, context, environment, and version
- **Context Generation**: Anonymizes sensitive data while preserving conversation essence
- **Error Handling**: Graceful fallbacks for missing sessions or context generation failures
- **Performance**: Async operations and caching minimize impact on user experience

### Testing

All tests pass with comprehensive coverage:
- Unit tests for command handler logic
- Context generation and caching behavior
- Database model updates
- Integration with existing message handlers
- Agent configuration and validation

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Added a /feedback workflow with a one-tap button to submit a prefilled feedback form and dynamic response handling.

* **Performance**
  * Occasional background pre-generation of feedback context to make feedback interactions snappier.

* **Privacy / Tracking**
  * Command usage tracking now accepts missing/anonymous session identifiers so analytics won't break when sessions are unavailable.

### CODE REVIEW

File: src/areyouok_telegram/data/models/command_usage.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class CommandUsage(Base):
 
     command = Column(String, nullable=False)
     chat_id = Column(String, nullable=False)
-    session_id = Column(String, nullable=False)
+    session_id = Column(String, nullable=True)
```
[the remark this answers is not in the document]
Comment (btimothy-har) [reply src/areyouok_telegram/data/models/command_usage.py:23]: Not necessary since this isn't yet pushed onto our main db.

File: src/areyouok_telegram/data/operations.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ async def track_command_usage(
     *,
     command: str,
     chat_id: str,
-    session_id: str,
+    session_id: str | None = None,
 ):
```
[the remark this answers is not in the document]
Comment (btimothy-har) [reply src/areyouok_telegram/data/operations.py:160]: Not necessary since this isn't yet pushed onto our main db.

File: src/areyouok_telegram/handlers/commands/feedback.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        feedback_context = await context_task
+
+        feedback_url = FEEDBACK_URL.format(
+            uuid=feedback_uuid,
+            session_id=active_session.session_id,
+            context=feedback_context,
+            env=ENV,
+            version=package_version(),
+        )
+
```
[the remark this answers is not in the document]
Comment (btimothy-har) [reply src/areyouok_telegram/handlers/commands/feedback.py:100]: Fixed in https://github.com/btimothy-har/areyouok-telegram/pull/134/commits/4d5d1342fc0cce32afac2d7362b8c0cdd5f50307 by hardening the agent output
